### PR TITLE
Refactor/calling sfx

### PR DESCRIPTION
**Pull request by DJSteins**
1. Added a new struct called PerigonSoundEffects that handles playing the sfx from a serialized audio clip and setting the audio group from a serialized property.
2. Updated all serialized audio clips to use PerigonSoundEffects.
3. Removed deprecated audio manager functions and sound effects enums

BUGFIX: Fixed the bug where the engine plays in the end game

### CODE REVIEW

**Comment by DJSteins** [comment]
[Refactor how SFX are being called (2.0)](https://trello.com/c/mWa7r7ST/754-refactor-how-sfx-are-being-called-20)

**Comment by DJSteins** [comment]
[The car engine sound can be heard when the game is paused (when going full speed) (0.5)](https://trello.com/c/WjxFj9RV/793-the-car-engine-sound-can-be-heard-when-the-game-is-paused-when-going-full-speed-05)

File: Assets/_PartyParkade/Script/Data/SoundEffects.cs [hunk 1]
```diff
@@ -5,27 +5,43 @@
...
+            if (obj is PerigonSoundEffect)
+            {
+                return obj.GetHashCode() == this.GetHashCode();
+            }
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
```
Comment (m4rtint) [inline Assets/_PartyParkade/Script/Data/SoundEffects.cs:44]: I don't think you need this method if you're just going to call the base method.

File: Assets/_PartyParkade/Script/Game/Car/CarAudioSystem.cs [hunk 2]
```diff
@@ -158,8 +158,7 @@ private void OnCollideWithObstacle(GameObject obstacle, float volume)
             {
                 if (AudioManager.Instance)
```
Comment (m4rtint) [inline Assets/_PartyParkade/Script/Game/Car/CarAudioSystem.cs:159]: You can probably cache this AudioManager.Instance at the top.

File: Assets/_PartyParkade/Script/ScriptableObject/SoundFX/InteractionsSFX.cs [hunk 3]
```diff
@@ -12,8 +14,8 @@ namespace _PartyParkade.Script.ScriptableObject
     [CreateAssetMenu(fileName = "Interactions", menuName = "SoundFX/Interactions")]
     public class InteractionsSFX : UnityEngine.ScriptableObject
     {
-        public AudioClip PlayerCollision = null;
-        public AudioClip ObstacleCollision = null;
-        public AudioClip Parked = null;
+        [ShowInInspector] public PerigonSoundEffect PlayerCollision;
```
Comment (m4rtint) [inline Assets/_PartyParkade/Script/ScriptableObject/SoundFX/InteractionsSFX.cs:17]: I thought making it a public variable already shows in Inspector?

Comment (DJSteins) [reply Assets/_PartyParkade/Script/ScriptableObject/SoundFX/InteractionsSFX.cs:17]: Oh yeah it does oops. I thought it wasn't working at first when I didn't use ShowInInspector but maybe it just hadn't updated.

**Comment by m4rtint** [comment]
`<<<<<<< refactor/calling_sfx
    CarHatPosition: {x: 0, y: 2.8, z: 0}
    Engine:
      AudioGroup: {fileID: 6667284257155179277, guid: 874fe6f8b30374b4cb7076373269123e,
        type: 2}
      AudioClip: {fileID: 8300000, guid: 6736ee64d28360d4b9eab806d700b1d9, type: 3}
      Loop: 1
    Horn:
      AudioGroup: {fileID: 1054175008289593493, guid: 874fe6f8b30374b4cb7076373269123e,
        type: 2}
      AudioClip: {fileID: 8300000, guid: a2dd67e4b2871264face4fa24b77fb26, type: 3}
      Loop: 0
=======
    CarHatPosition: {x: 0, y: 2.3, z: 0}
    Engine: {fileID: 8300000, guid: 6736ee64d28360d4b9eab806d700b1d9, type: 3}
    Horn: {fileID: 8300000, guid: a2dd67e4b2871264face4fa24b77fb26, type: 3}
>>>>>>> develop`


Can keep your section, just make sure the carHatPosition is 2.3

**Pull request merged**